includes/polling/os/ucos.inc.php

```diff
@@ -11,3 +11,21 @@
  */
 
 $hardware = snmp_get($device, 'sysObjectID.0', '-Osqv', 'SNMPv2-MIB:CISCO-PRODUCTS-MIB');
+
+$version = snmp_get($device, '.1.3.6.1.2.1.54.1.1.1.1.4.1', '-Osqv');
+
+$applistraw = snmpwalk_cache_oid($device, ".1.3.6.1.2.1.54.1.1.1.1.3", array());
```

Comment:
You could actually just do a regular snmpwalk: `snmp_walk($device, '.1.3.6.1.2.1.54.1.1.1.1.3', '-OQv')`
Then use text tools like str_contains() or preg_match() to check for strings.

Reply:
Fixed, Much better, thanks!